helper-tools/prepare-jamf-policy.sh

```diff
@@ -0,0 +1,92 @@
+#!/bin/bash
+#-
+#- Usage
+#-   $ ./prepare-jamf-policy.sh "/path/to/Install macOS.app"
+#-   Wait seconds, getting checksumm...
+#-
+#-   =====================================================
+#-   Upgrade macOS Parameters:
+#-   Parameter 4: /Applications/Install macOS.app
+#-   Parameter 5: 10.14
+#-   Parameter 6: (Your download trigger policy)
+#-   Parameter 7: 20acadc8d66bb3d882bc0fedf1358c64
+#-   =====================================================
+#-
+#-   Would you need a package archive of Install macOS Mojave.app? [y/n]: y
+#-   Ok, building package archive file of Install macOS Mojave.app. Wait few minutes.
+#-   Done. ~/Desktop/Install macOS Mojave.app.10.14.pkg
+#-   $
+#-
+#-   You can get full-size installer with installinstallmacos.py.
+#-   https://github.com/munki/macadmin-scripts/blob/master/installinstallmacos.py
+#-
+
+if [ $# -eq 0 ]; then
+    # Show help
+    /usr/bin/grep ^#- "$0" | cut -c 4-
+    exit 0
+fi
+
+OSInstaller="$1"
+if [ ! -d "$OSInstaller" ]; then
+    echo "Not found $OSInstaller !"
+    exit 1
+fi
+
+if [ ! -d  "${OSInstaller}/Contents/SharedSupport" ]; then
+    echo "This installer looks like kind of 'stub' installer."
+    /usr/bin/du -sh "${OSInstaller}"
+    echo "Use full size intaller."
+    echo "You can get full-size installer with installinstallmacos.py"
+    echo "https://github.com/munki/macadmin-scripts/blob/master/installinstallmacos.py"
+    exit 1
+fi
+
+if [ ! -f  "${OSInstaller}/Contents/SharedSupport/InstallInfo.plist" ]; then
+    echo "Not found ${OSInstaller}/Contents/SharedSupport/InstallInfo.plist"
+    echo "Unknown installer type. Apple may change something."
+    exit 1
+fi
+
+if [ ! -f  "${OSInstaller}/Contents/SharedSupport/InstallESD.dmg" ]; then
+    echo "Not found ${OSInstaller}/Contents/SharedSupport/InstallESD.dmg"
+    echo "Thi is not expected installer type."
+    exit 1
+fi
+
+osversion=$(/usr/libexec/PlistBuddy -c "print 'System Image Info:version'" "${OSInstaller}/Contents/SharedSupport/InstallInfo.plist")
+echo "Wait seconds, getting checksumm..."
+checksum=$(/sbin/md5 -r "${OSInstaller}/Contents/SharedSupport/InstallESD.dmg" | /usr/bin/awk '{print $1}')
+
+cat <<_RESULT
+
+=====================================================
+Parameters for JamfPro policy:
+Parameter 4: /Applications/$( basename "$OSInstaller" )
+Parameter 5: $osversion
+Parameter 6: (Your download trigger policy)
+Parameter 7: $checksum
+=====================================================
+
+_RESULT
+
+# Build a package of installer
```

Comment:
Some people want to sign package builds.
So I don't think you should do it inside this script.

Reply:
In that case, user can type 'n' or anything other than 'y'.

Reply:
Yes. However, if you allow this, it will guide you to create reads that answer y/n one after another when you need other options, so I think we should limit the functionality as much as possible.

Reply:
The main purpose of this script is to help macOSUpgrade.sh with Jamfpro.
User need both a installer info and installer itself. 
I believe it is reasonable this script will deal both. 
In this case, asking user if create package is not strange and I will request if this script did not have such kind of choice . 

I think it is no problem  some options will be asked and other ones won't.

Reply:
I understand.
I do not oppose with a strong will. We think that it is good to adopt this.